analytics/urls.py

```diff
@@ -1,33 +1,35 @@
-from django.conf.urls import include, url
+from django.conf.urls import include
+from django.urls import path
+
 
 import analytics.views
 from zerver.lib.rest import rest_dispatch
 
 i18n_urlpatterns = [
     # Server admin (user_profile.is_staff) visible stats pages
-    url(r'^activity$', analytics.views.get_activity,
-        name='analytics.views.get_activity'),
-    url(r'^activity/support$', analytics.views.support,
-        name='analytics.views.support'),
-    url(r'^realm_activity/(?P<realm_str>[\S]+)/$', analytics.views.get_realm_activity,
-        name='analytics.views.get_realm_activity'),
-    url(r'^user_activity/(?P<email>[\S]+)/$', analytics.views.get_user_activity,
-        name='analytics.views.get_user_activity'),
-
-    url(r'^stats/realm/(?P<realm_str>[\S]+)/$', analytics.views.stats_for_realm,
-        name='analytics.views.stats_for_realm'),
-    url(r'^stats/installation$', analytics.views.stats_for_installation,
-        name='analytics.views.stats_for_installation'),
-    url(r'^stats/remote/(?P<remote_server_id>[\S]+)/installation$',
-        analytics.views.stats_for_remote_installation,
-        name='analytics.views.stats_for_remote_installation'),
-    url(r'^stats/remote/(?P<remote_server_id>[\S]+)/realm/(?P<remote_realm_id>[\S]+)/$',
-        analytics.views.stats_for_remote_realm,
-        name='analytics.views.stats_for_remote_realm'),
+    path('activity', analytics.views.get_activity,
+         name='analytics.views.get_activity'),
+    path('activity/support', analytics.views.support,
+         name='analytics.views.support'),
+    path('realm_activity/<str:realm_str>/', analytics.views.get_realm_activity,
+         name='analytics.views.get_realm_activity'),
+    path('user_activity/<str:email>/', analytics.views.get_user_activity,
+         name='analytics.views.get_user_activity'),
+
+    path('stats/realm/<str:realm_str>/', analytics.views.stats_for_realm,
+         name='analytics.views.stats_for_realm'),
+    path('stats/installation', analytics.views.stats_for_installation,
+         name='analytics.views.stats_for_installation'),
+    path('stats/remote/<str:remote_server_id>/installation',
+         analytics.views.stats_for_remote_installation,
+         name='analytics.views.stats_for_remote_installation'),
+    path('stats/remote/<str:remote_server_id>/realm/<str:remote_realm_id>/',
```

Comment:
Should `remote_realm_id` be tagged as int rather than str?  I think that's its format.

Reply:
Hmm, [\S] which was previously fooled me, but yeah, stats_for_remote_realm takes int as an argument, and the same thing is with the `remote_server_id`.
But I'm not sure, what can we do with other urls with `remote_realm_id` and `remote_server_id`? They pass that argument to `rest_dispatch`, and I guess we can't know how it will be used (as an int or str). Should we set them both int, too? Let me know, and I will change that tomorrow ASAP.

Reply:
Well, `remote_server_id`s are UUIDs aka strings, and I assume we'll just validate as we have been in the view function.  So it's really just `remote_realm_id` here where we can do something better.

Reply:
It might be better to just convert everything to `str:` for now; we might need to change tests for other changes because the error handling of this Django feature might be different from ours.